ecl/hql/hqlgram2.cpp

```diff
@@ -8992,6 +8992,34 @@ bool HqlGram::convertAllToAttribute(attribute &atr)
     return true;
 }
 
+void HqlGram::setPluggableModeExpr(attribute & targetAttr, attribute & pluginAttr, attribute & options)
+{
+    IAtom * fileFormatAtom = lower(pluginAttr.getId());
+    StringBuffer fileFormat(fileFormatAtom->queryStr());
+    const char * fileFormatStr = fileFormat.str();
+    bool isSupportedFileType = false;
+
+    // TODO: Hardcoded tests for known filetype plugins; should be
+    // an iteration through a list of available types, checking for names
+    isSupportedFileType = strsame(fileFormatStr, "parquet");
+
+    if (isSupportedFileType)
+    {
+        // Do we just cite the name of the plugin, or perhaps something
+        // like a unique ID from the plugin's factory (assuming there is one)?
+        OwnedHqlExpr fileTypeExp = createExprAttribute(fileTypeAtom, createConstant(fileFormatStr));
+        HqlExprArray args;
+        options.unwindCommaList(args);
+        // Make sure the filetype is first in the args list
+        args.add(*LINK(fileTypeExp), 0);
+        targetAttr.setExpr(createValue(no_filetype, makeNullType(), args));
+    }
+    else
+    {
+        // ERR_UNKNOWN_TYPE is generic; create file type-specific error?
+        reportError(ERR_UNKNOWN_TYPE, pluginAttr, "Unknown plugin file type %s", fileFormatStr);
+    }
```

Comment:
Best to still set targetAttr() to something innocuous.  

E.g. targetAttr.setExpr(createValue(no_thor, makeNullType())

so that any further productions always get back an expression.